src/test/java/io/cdap/plugin/servicenow/connector/ServiceNowConnectorTest.java

```diff
@@ -0,0 +1,178 @@
+    result.add(map);
+    int httpStatus = HttpStatus.SC_OK;
+    Map<String, String> headers = new HashMap<>();
+    String responseBody = "{\n" +
+      "    \"result\": [\n" +
+      "        {\n" +
+      "            \"calendar_integration\": \"1\",\n" +
+      "            \"date_format\": \"\",\n" +
+      "            \"location\": \"\"\n" +
+      "        }\n" +
+      "    ]\n" +
+      "}";
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    Mockito.when(restApi.executeGet(Mockito.any())).thenReturn(restAPIResponse);
+    Mockito.when(restApi.parseResponseToResultListOfMap(restAPIResponse.getResponseBody())).thenReturn(result);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+      withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
+      thenReturn(accessTokenResponse);
+    Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(ArgumentMatchers.any(), ArgumentMatchers.anyString())).
+      thenReturn(response);
+    ServiceNowConnector serviceNowConnector = new ServiceNowConnector(serviceNowSourceConfig.getConnection());
+    serviceNowConnector.test(context);
+    ServiceNowTableInfo serviceNowTableInfo = new ServiceNowTableInfo("table", getPluginSchema(), 1);
+    List<ServiceNowTableInfo> list = new ArrayList<>();
+    list.add(serviceNowTableInfo);
+    PowerMockito.mockStatic(ServiceNowInputFormat.class);
+    SourceQueryMode mode = SourceQueryMode.TABLE;
+    SourceValueType valueType = SourceValueType.SHOW_DISPLAY_VALUE;
+    Mockito.when(ServiceNowInputFormat.fetchTableInfo(mode, serviceNowSourceConfig.getConnection(),
+                                                      serviceNowSourceConfig.getTableName(),
+                                                      null, valueType,
+                                                      null, null)).thenReturn(list);
+
+    ConnectorSpec connectorSpec = serviceNowConnector.generateSpec(new MockConnectorContext
```

Comment:
We should have separate test cases for each method that's being tested.
Split each test case into 3 sections:
- Test setup (create mocks, objects required to execute the test case)
- Method that you are testing. e.g: serviceNowConnector.generateSpec()
- Validate that the test did what we expected. Assert that methods were called, failures added in collector etc.

Reply:
done